src/main/java/Duke.java

```diff
@@ -1,10 +1,319 @@
+        String input = sc.nextLine();
+        if (input.equals("yes") || input.equals("y")) {
+            if (word.equals("all")) {
+                list.clear();
+                System.out.println("List is now empty.");
+            } else {
+                List<Task> newList = new ArrayList<>();
+                for (Task t : list) {
+                    if (!t.isDone()) {
+                        newList.add(t);
+                    }
+                }
+                list = newList;
+                System.out.println("List cleared.");
+                printList(list);
+            }
+            FileWriter writer = new FileWriter("data/duke.txt");
+            for (Task t : list) {
+                writer.write(t.toSimplerString() + "\n");
+            }
+            writer.close();
+        } else {
+            System.out.println("Canceled.");
+        }
+    }
+
+    private static void addEventToList(List<Task> list, String line) {
+        String[] lineByWord = line.split("//");
+        boolean isDone;
+        isDone = lineByWord[1].equals("T");
+        switch (lineByWord[0]) {
+        case "T":
+            list.add(new Todo(isDone, lineByWord[2]));
+            break;
+        case "D":
+            list.add(new Deadline(isDone, lineByWord[2], LocalDate.parse(lineByWord[3])));
+            break;
+        case "E":
+            list.add(new Event(isDone, lineByWord[2], LocalDate.parse(lineByWord[3])));
+            break;
+        }
+    }
+
+    private static void writeToHardDisk(List<Task> list) throws IOException {
+        FileWriter writer = new FileWriter("data/duke.txt");
+        for (Task t : list) {
+            writer.write(t.toSimplerString() + "\n");
+        }
+        writer.close();
+    }
+
+    static Task createAnEventTask(String[] instructionByWord, int lengthOfArray) throws DukeException {
+        try {
+            int indexOfAt = -1;
+            for (int i = 1; i < lengthOfArray; i++) {
+                if (instructionByWord[i].equals("/at")) {
+                    indexOfAt = i;
+                }
+            }
+            if (indexOfAt == -1 || indexOfAt == 1 || indexOfAt == lengthOfArray) {
+                throw new DukeException(FORMAT_CORRECTION
+                        + "\"event a_string_describing_the_task /at YYYY-MM-DD\"");
+            }
+            String description = String.join(" ",
+                    Arrays.copyOfRange(instructionByWord, 1, indexOfAt));
+            String time = String.join(" ",
+                    Arrays.copyOfRange(instructionByWord, indexOfAt + 1, lengthOfArray));
+            return new Event(description, LocalDate.parse(time));
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Incorrect format of date.\n"
+                    + "The correct format should be YYYY-MM-DD.");
+        } catch (DateTimeException e) {
+            throw new DukeException("Invalid value for year/month/date");
```

Comment:
Yes I agree you can directly print out the message from there. I would think that it would be better if you only throw DukeExceptions when certain things are not the way they should be in context of Duke itself.

Reply:
I was trying to handle two types of erroneous user input separately. The first case is that the user does not input a valid format for the date, the second one is that the format is valid, but the date is not (for example 2020-02-31).